vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/ExperimentInformationPage.java

```diff
@@ -47,72 +42,71 @@
         System.identityHashCode(this), System.identityHashCode(projectNavigationBarComponent),
-        System.identityHashCode(experimentDetailsComponent)
-      ));
+        System.identityHashCode(experimentMainComponent),
+        System.identityHashCode(experimentSupportComponent))
+    );
   }
 
   private void setupBoard(ProjectNavigationBarComponent projectNavigationBarComponent,
-      ExperimentDetailsComponent experimentDetailsComponent) {
-    Board board = new Board();
-
-    Row rootRow = new Row();
-
-    VerticalLayout mainComponents = new VerticalLayout();
-    mainComponents.setPadding(false);
-    mainComponents.setMargin(false);
-    mainComponents.setSpacing(false);
-    mainComponents.add(projectNavigationBarComponent, experimentDetailsComponent);
-
-    rootRow.add(mainComponents, 3);
-    itemCollection = ExperimentItemCollection.create("Add a new experiment");
-    rootRow.add(itemCollection, 1);
-
-    board.add(rootRow);
-
-    board.setSizeFull();
-
-    add(board);
+      ExperimentMainComponent experimentMainComponent,
+      ExperimentSupportComponent experimentSupportComponent) {
+    this.addClassName("experiment-page");
+    this.add(projectNavigationBarComponent);
+    this.add(experimentMainComponent);
+    this.add(experimentSupportComponent);
   }
 
+  /**
+   * Extracts {@link ExperimentId} from the provided URL before the user accesses the page
+   * <p>
+   * This method is responsible for checking if the provided {@link ExperimentId} is valid and
+   * triggering its propagation to the components within the {@link ExperimentInformationPage}
+   */
+
   @Override
   public void beforeEnter(BeforeEnterEvent beforeEnterEvent) {
-    beforeEnterEvent.getRouteParameters().get(EXPERIMENT_ID_ROUTE_PARAMETER).ifPresent(experimentIdParam -> {
-      try {
-        ExperimentId experimentId = ExperimentId.parse(experimentIdParam);
-        experimentInformationPageHandler.setExperimentId(experimentId);
-      } catch (IllegalArgumentException e) {
-        log.debug(String.format("Provided ExperimentId %s is invalid due to %s", experimentIdParam,
-            e.getMessage()));
-        experimentInformationPageHandler.rerouteToActiveExperiment(beforeEnterEvent);
-      }
-    });
+    beforeEnterEvent.getRouteParameters().get(EXPERIMENT_ID_ROUTE_PARAMETER)
+        .ifPresent(experimentIdParam -> {
+          try {
+            ExperimentId experimentId = ExperimentId.parse(experimentIdParam);
+            experimentInformationPageHandler.setExperimentId(experimentId);
+          } catch (IllegalArgumentException e) {
+            log.debug(
+                String.format("Provided ExperimentId %s is invalid due to %s", experimentIdParam,
+                    e.getMessage()));
+            experimentInformationPageHandler.rerouteToActiveExperiment(beforeEnterEvent);
```

Comment:
for readability, I suggest to extract that in an own method

Reply:
done 👍